spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -113,16 +116,40 @@ public CloudSqlJdbcInfoProvider appengineCloudSqlJdbcInfoProvider(
 
 	@Bean
 	@ConditionalOnMissingBean(CloudSqlJdbcInfoProvider.class)
-	public CloudSqlJdbcInfoProvider cloudSqlJdbcInfoProvider(SQLAdmin sqlAdmin,
+	@ConditionalOnProperty(value = "spring.cloud.gcp.sql.database-type", havingValue = "postgresql")
+	public CloudSqlJdbcInfoProvider postgreSqlJdbcInfoProvider(SQLAdmin sqlAdmin,
 			GcpProjectIdProvider projectIdProvider) {
 		CloudSqlJdbcInfoProvider defaultProvider = new DefaultCloudSqlJdbcInfoProvider(
 				projectIdProvider.getProjectId(),
 				sqlAdmin,
-				this.gcpCloudSqlProperties);
+				this.gcpCloudSqlProperties,
+				DatabaseType.POSTGRESQL);
```

Comment:
Something smells here. We have now `public DatabaseType getDatabaseType() `, so `DefaultCloudSqlJdbcInfoProvider` can easily determine what to do or even it won't care because `DatabaseType` does the magic already.
IMO we need only one `DefaultCloudSqlJdbcInfoProvider` bean and rely on the provided properties, without any `@ConditionalOnProperty`

Reply:
Yes, good catch!